src/Microsoft.AspNet.FileProviders.Combined/CombinedFileProvider.cs

```diff
@@ -0,0 +1,205 @@
+                    var directories =new List<IDirectoryContents>();
+                    foreach (var fileProvider in fileProviders)
+                    {
+                        var directoryContents = fileProvider.GetDirectoryContents(subpath);
+                        if (directoryContents != null && directoryContents.Exists)
+                        {
+                            directories.Add(directoryContents);
+                        }
+                    }
+                    return directories;
+                }
+                );
+
+                _files = new Lazy<List<IFileInfo>>(() =>
+                {
+                    var files = new List<IFileInfo>();
+                    var names = new HashSet<string>();
+
+                    var directories = _directoriesContents.Value;
+                    for (int i = 0; i < directories.Count; i++)
+                    {
+                        var directoryContents = directories[i];
+                        foreach (var file in directoryContents)
+                        {
+                            if (names.Add(file.Name))
+                            {
+                                files.Add(file);
+                            }
+                        }
+                    }
+                    return files;
+                });
+
+                _exists = new Lazy<bool>(() =>
+                {
+                    var directories = _directoriesContents.Value;
+                    var exists = directories.Count > 0;
+                    return exists;
+                });
+            }
+
+            public IEnumerator<IFileInfo> GetEnumerator()
+            {
+                return _files.Value.GetEnumerator();
+            }
+
+            IEnumerator IEnumerable.GetEnumerator()
+            {
+                return _files.Value.GetEnumerator();
+            }
+
+            // This directory exists because it is created only when there is existing IDrectoryContents to merge.
+            public bool Exists { get { return _exists.Value; } }
+        }
+
+        private class CombinedFileChangeToken : IChangeToken
+        {
+            private readonly List<IChangeToken> _changeTokens;
+
+            public CombinedFileChangeToken(List<IChangeToken> changeTokens)
+            {
+                _changeTokens = changeTokens;
+            }
+
+            public IDisposable RegisterChangeCallback(Action<object> callback, object state)
+            {
+                var disposables = new List<IDisposable>();
+                for (int i = 0; i < _changeTokens.Count; i++)
+                {
+                    var disposable = _changeTokens[i].RegisterChangeCallback(callback, state);
+                    disposables.Add(disposable);
+                }
+                return new Disposables(disposables);
+            }
+
+            public bool HasChanged
+            {
+                get { return _changeTokens.Any(_ => _.HasChanged); }
```

Comment:
Don't use underscores for things you're using: 
`_changeTokens.Any(token => token.HasChanged)`. Same comment for `ActiveChangeCallbacks`